monitoring/slack_alerts/ecs_tasks_cant_start_alert/src/ecs_tasks_cant_start_alert.py

```diff
@@ -0,0 +1,101 @@
+import functools
+import json
+import os
+import sys
+import urllib.request
+
+import boto3
+
+
+def log_on_error(fn):
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except Exception:
+            print(f"args   = {args!r}", file=sys.stderr)
+            print(f"kwargs = {kwargs!r}", file=sys.stderr)
+            raise
+
+    return wrapper
+
+
+def get_ecs_events(sess, *, cluster_arn, service_name):
+    """
+    Return the events for a given ECS service name.  These are the
+    events shown in the "Events" tab of the ECS console.
+    """
+    client = sess.client("ecs")
+    resp = client.describe_services(cluster=cluster_arn, services=[service_name])
+
+    return resp["services"][0]["events"]
+
+
+@functools.cache
+def get_secret_string(sess, *, secret_id):
```

Comment:
I vaguely recall that we do/did this somewhere already but it does feel like a package of weco utils and other stuff that we use all over the place on PyPI would be useful

Reply:
We do this all over the place, this function is very common.

We have done that in the past (see https://github.com/wellcomecollection/aws_utils); I've deliberately eschewed it here because it makes it significantly harder to deploy and debug your Lambdas.